crates/cli/commands/src/p2p/rlpx.rs

```diff
@@ -0,0 +1,54 @@
+//! RLPx subcommand of P2P Debugging tool.
+
+use clap::{Parser, Subcommand};
+use discv5::Enr;
+use reth_discv5::enr::EnrCombinedKeyWrapper;
+use reth_ecies::stream::ECIESStream;
+use reth_eth_wire::{HelloMessage, UnauthedP2PStream};
+use reth_network::config::rng_secret_key;
+use reth_network_peers::{pk2id, NodeRecord};
+use secp256k1::SECP256K1;
+use tokio::net::TcpStream;
+
+/// RLPx commands
+#[derive(Parser, Debug)]
+pub struct Command {
+    #[clap(subcommand)]
+    subcommand: Subcommands,
+}
+
+impl Command {
+    // Execute `p2p rlpx` command.
+    pub async fn execute(self) -> eyre::Result<()> {
+        match self.subcommand {
+            Subcommands::Ping { node } => {
+                let key = rng_secret_key();
+                let enr = node.parse::<Enr>().unwrap();
+                let enr = EnrCombinedKeyWrapper(enr).into();
+                let node_record = NodeRecord::try_from(&enr)?;
+                let outgoing =
+                    TcpStream::connect((node_record.address, node_record.tcp_port)).await?;
+                let ecies_stream = ECIESStream::connect(outgoing, key, node_record.id).await?;
+
+                let peer_id = pk2id(&key.public_key(SECP256K1));
+                let hello = HelloMessage::builder(peer_id).build();
+
+                let (_, their_hello) =
+                    UnauthedP2PStream::new(ecies_stream).handshake(hello).await?;
+
+                println!("{:#?}", their_hello);
```

Comment:
> this should follow on to use the established session returned by the handshake (the first index in the tuple), and use that to send a ping

@mattsse I try to implement this which need to refactor some core code of p2p, it may need some time. Or we could merge this first and I'll implement `send ping` in another PR :)

Reply:
cool, let's get this over the line and followup separately

Reply:
please change the name of the pr then, since it only partially implements the ping command